// log4jse/index.js
/**
 * Log4Jse
 * 		A log util for JavaScript and Node.js
 * @author ijse
 * @date 2012-05-23
 * @version  0.2.0
 */
//TODO: Finish the Log Util
function extend(target, options, deep) {
    for (name in options) {
        copy = options[name];
        if (deep && copy instanceof Array) {
            target[name] = $.extend(deep, [], copy);
        } else if (deep && copy instanceof Object) {
            target[name] = $.extend(deep, {}, copy);
        } else {
            target[name] = options[name];
        }
    }
    return target;
}

var Logger = function() {
	var LEVELS = ["OFF","FATAL","ERROR","WARN","INFO","DEBUG","ALL"];
	var defaultOptions = {
		appName: "LOG4JSE",
		level: "ALL", // 6 is ok too
		format: [
			{
				"for": ["DEBUG", "INFO"], //
				"template": "[%LEVEL%][%APPNAME%]%DATE(yyyy-MM-dd HH:mm:ss)%\t%CONTENT%"
			}
		],
		output: [
			{
				"for": ["DEBUG", "INFO"], // "DEBUG" for all beyond debug level
				"fn" :function(msg, msgObj) {
					console.ltog(msg);
				}
			}
		]
	}
	var level = LEVELS.indexOf("ALL"); // Default ALL
}

exports.getLogger = function() {

}
exports.option = function() {

}
